backends/events/default/default-events.cpp

```diff
@@ -195,10 +195,13 @@ bool DefaultEventManager::pollEvent(Common::Event &event) {
 
 			{
 				PauseToken pt;
-				if (g_engine)
+				if (g_engine) {
 					pt = g_engine->pauseEngine();
-				GUI::MessageDialog alert(_("Do you really want to quit?\nAny unsaved progress will be lost."), _("Quit"), _("Cancel"));
-				forwardEvent = _shouldQuit = (alert.runModal() == GUI::kMessageOK);
+					GUI::MessageDialog alert(_("Do you really want to quit?\nAny unsaved progress will be lost."), _("Quit"), _("Cancel"));
+					forwardEvent = _shouldQuit = (alert.runModal() == GUI::kMessageOK);
+				} else {
+					_shouldQuit = true;
+				}
```

Comment:
That change is fine with me, and this makes quitting from the launcher consistent whatever method is used to achieve it (Quit button in Launcher, Quit from menu on macOS, or the hotkey). One of the reason for this dialog though was to prevent quitting by mistake when hitting the wrong hotkey. But if you are in the launcher you are not going to lose any progress anyway, so quitting by mistake is not a huge issue.

However you could reorganise the code for the `Common::EventQuit` case to simplify it and make it more easy to read:
```c++
if (g_engine && ConfMan.getBool("confirm_exit")) {
	if (_confirmExitDialogActive) {
		forwardEvent = false;
		break;
	}
	_confirmExitDialogActive = true;
	{
	 	PauseToken pt =  g_engine->pauseEngine();
	 	GUI::MessageDialog alert(_("Do you really want to quit?\nAny unsaved progress will be lost."), _("Quit"), _("Cancel"));
	 	forwardEvent = _shouldQuit = (alert.runModal() == GUI::kMessageOK);
	}
	_confirmExitDialogActive = false;
} else {
	_shouldQuit = true;
}
```

Reply:
Thanks, this is obviously much cleaner than what I did :-)

Changed with ce4375e.